src/validate_pyproject/extra_validations.py

```diff
@@ -49,4 +56,28 @@ def validate_project_dynamic(pyproject: T) -> T:
     return pyproject
 
 
-EXTRA_VALIDATIONS = (validate_project_dynamic,)
+def validate_include_depenency(pyproject: T) -> T:
+    dependency_groups = pyproject.get("dependency-groups", {})
+    for key, value in dependency_groups.items():
+        for each in value:
+            if (
+                isinstance(each, dict)
+                and (include_group := each.get("include-group"))
+                and include_group not in dependency_groups
+            ):
+                raise IncludedDependencyGroupMustExist(
+                    message=f"The included dependency group {include_group} doesn't exist",
+                    value={
+                        "field": include_group,
+                    },
```

Comment:
```suggestion
                    value={"include-group": include_group},
                    # Alternatively: value=each
```

As far as I remember the `value` field corresponds to the given `pyproject.toml` excerpt converted to JSON. The idea is to tell the user which configuration is wrong...

Reply:
Do you mean

```suggestion
                    value={
                        "value": each,
                    },
```
?

Reply:
When proposing the second alternative, I was thinking in

```suggestion
                    value=each,
```

`each` is the part of the configuration where the fail happens, right? (i.e. the part that contains the wrong `include-group`?)

So if we do `value=each` the exception will give the user some context.
If we do `value={"include-group": include_group}` then the exception error will show only the specific part that is causing the error.

There are many variations of this depending on the amount of context we want to show the user about the error.